src/RiotAPILibraries.ts

```diff
@@ -0,0 +1,167 @@
+            }
+        }
+    }
+
+    onBot() {
+        console.log("Github extension loaded.");
+    }
+
+    async onCommand(message: Discord.Message) {
+
+        const args = message.content.split(" ");
+
+        // if it's just our command, get the list
+        if (args.length < 2) {
+            args.push("list");
+        }
+
+        const [command, language] = args;
+
+        // if !libs, !libraries, etc
+        if (this.settings.githubLibraries.aliases.some(x => x === command)) {
+
+            if (language === "list") {
+                const response = await fetch(this.settings.githubLibraries.baseURL, this.fetchSettings);
+                const data = await response.json() as GithubAPIStruct[];
+
+                let languages = "`" + data.map(x => x.name).join(", ") + "`";
+                let reply = this.settings.githubLibraries.languageList.replace("{languages}", languages);
+                message.channel.send(reply);
+                return;
+            }
+
+            const response = await fetch(this.settings.githubLibraries.baseURL + language);
+            if (response.status != 200) {
+                message.channel.send(this.settings.githubLibraries.githubError + response.status);
+                return;
+            }
+            
+            const libraryList = await response.json();
+            if (!Array.isArray(libraryList) || libraryList.length === 0 || !libraryList[0].sha) {
+                message.channel.send(this.settings.githubLibraries.noLanguage + language);
+                return;
+            }
+
+            let editMessagePromise = message.channel.send(`Found the list of libraries for ${language}, listing ${libraryList.length} libraries, this post will be edited with the result.`);
+
+            let promises: Promise<LibraryDescription>[] = [];
+            for (const library of libraryList) {
+                promises.push(this.describeAPILibrary(library));
+            }
+
+            const libraryDescriptions = (await Promise.all(promises))
+            .filter(l => l.valid && l.library) // Only valid ones
+            .sort((a, b) => b.stars - a.stars); // Sort by stars
+        
+            const embed = new Discord.RichEmbed({ title: `List of libraries for ${language}:` });
+            for (const desc of libraryDescriptions) {
+                if (!desc.library) continue; // https://github.com/Microsoft/TypeScript/issues/18562
+                embed.addField(`${desc.library.repo} (⭐ ${desc.stars ? desc.stars : "0"})`, `${desc.library.description ? desc.library.description + "\n" : " "}${desc.links.join(", ")}`);
+            }
+
+            let editMessage = await editMessagePromise; 
+            if (Array.isArray(editMessage)) editMessage = editMessage[0];
+            editMessage.edit({ embed });
+        }
+    }
+
+    async describeAPILibrary(json: GithubAPIStruct): Promise<LibraryDescription> {
+
+        return new Promise<LibraryDescription>(async(resolve, reject) => {
```

Comment:
Uhh. Async functions already return promises implicitly. You can remove this entirely and just replace `resolve` with a `return` and `reject` with a `throw`.